lib/rules/arrow-body-style.js

```diff
@@ -65,38 +68,57 @@ module.exports = {
             const arrowBody = node.body;
 
             if (arrowBody.type === "BlockStatement") {
-                if (never) {
+                const blockBody = arrowBody.body;
+
+                if (blockBody.length !== 1 && !never) {
+                    return;
+                }
+
+                if (asNeeded && requireReturnForObjectLiteral && blockBody[0].type === "ReturnStatement" &&
+                    blockBody[0].argument.type === "ObjectExpression") {
+                    return;
+                }
+
+                if (never || asNeeded && blockBody[0].type === "ReturnStatement") {
                     context.report({
                         node,
                         loc: arrowBody.loc.start,
-                        message: "Unexpected block statement surrounding arrow body."
-                    });
-                } else {
-                    const blockBody = arrowBody.body;
-
-                    if (blockBody.length !== 1) {
-                        return;
-                    }
+                        message: "Unexpected block statement surrounding arrow body.",
+                        fix(fixer) {
+                            if (blockBody.length !== 1 || blockBody[0].type !== "ReturnStatement") {
+                                return null;
+                            }
 
-                    if (asNeeded && requireReturnForObjectLiteral && blockBody[0].type === "ReturnStatement" &&
-                        blockBody[0].argument.type === "ObjectExpression") {
-                        return;
-                    }
+                            const returnValue = blockBody[0].argument;
+                            const sourceText = sourceCode.getText();
+                            const textBeforeReturn = sourceText.slice(arrowBody.range[0] + 1, blockBody[0].range[0]);
+                            const textBetweenReturnAndValue = sourceText.slice(sourceCode.getFirstToken(blockBody[0]).range[1], returnValue.range[0]);
+                            const returnValueText = returnValue.type === "ObjectExpression" ? `(${sourceCode.getText(returnValue)})` : sourceCode.getText(returnValue);
```

Comment:
This might cause a syntax error after autofix. It needs parentheses if the first token of `returnValue` is `{`.

```js
key => {
    return {a: 1, b: 2}[key]
}
```